internal/test/goleak/goleak_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) MongoDB, Inc. 2024-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/goleak"
+)
+
+var dbName = fmt.Sprintf("goleak-%d", time.Now().Unix())
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
+
+// TestGoroutineLeak creates clients with various client configurations, runs
+// some operations with each one, then disconnects the client. It asserts that
+// no goroutines were leaked after the client is disconnected.
+func TestGoroutineLeak(t *testing.T) {
+	uri := "mongodb://localhost:27017"
+	if u := os.Getenv("MONGODB_URI"); u != "" {
+		uri = u
+	}
+
+	testCases := []struct {
+		desc string
+		opts *options.ClientOptions
+	}{
+		{
+			desc: "base",
+			opts: options.Client().ApplyURI(uri),
```

Comment:
`ApplyURI` is used in every test and notably if it's empty, then it will default to `localhost:27017`. Suggest applying this value in the test runner:

```go
opts := tc.opts
if u := os.Getenv("MONGODB_URI"); u != "" {
	opts.ApplyURI(u)
}

client, err := mongo.Connect(context.Background(), opts)
require.NoError(t, err)
```

Reply:
My concern with that approach is that `ApplyURI` will overwrite the explicitly configured values if it's called after. However, I believe I can get the same effect using something like
```go
base := options.Client()
if u := os.Getenv("MONGODB_URI"); u != "" {
	base.ApplyURI(u)
}
client, err := mongo.Connect(context.Background(), base, t.opts)
```

Reply:
Updated.